tools/nl/analysis/log/mine.py

```diff
@@ -0,0 +1,186 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import csv
+from datetime import datetime
+from datetime import timedelta
+import json
+import logging
+
+from absl import app
+from absl import flags
+from google.cloud import bigtable
+from google.cloud.bigtable import row_filters
+
+FLAGS = flags.FLAGS
+
+flags.DEFINE_string('project', 'datcom-store', 'GCP project')
+flags.DEFINE_string('instance', 'website-data', 'BT instance')
+flags.DEFINE_string('table', 'nl-query', 'BT table')
+flags.DEFINE_integer('past_days', 5, 'Days to go back')
+flags.DEFINE_string('output_csv', '/tmp/nl_query_log.csv', 'Output CSV')
```

Comment:
name the csv file with the mined date "2023.07.01-2023.07.05***.csv"

Reply:
good idea.